analytic_engine/src/table/data.rs

```diff
@@ -86,6 +88,36 @@ impl TableShardInfo {
     }
 }
 
+/// The context for execution of serial operation on the table.
+pub struct SerialExecContext {
+    /// Denotes whether `serial_exec` is valid.
+    ///
+    /// The `serial_exec` will be invalidated if the table is closed.
+    invalid: bool,
+    serial_exec: TableOpSerialExecutor,
+}
+
+impl SerialExecContext {
+    #[inline]
+    pub fn invalidate(&mut self) {
+        self.invalid = true;
+    }
+}
+
+impl Deref for SerialExecContext {
+    type Target = TableOpSerialExecutor;
+
+    fn deref(&self) -> &Self::Target {
```

Comment:
It seems this method will leak `serial_exec`, which could only be used when invalid is false.

Reply:
Actually, the context can't be acquired if it is invalid.